### PR TITLE
Dbeaver/pro#2676 fix fqn view postgres

### PR DESCRIPTION
**Pull request by HocKu7**

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java [hunk 1]
```diff
@@ -122,6 +122,10 @@ public static String generateObjectDDL(@NotNull DBRProgressMonitor monitor, @Not
 
     public static String getTableDDL(@NotNull DBRProgressMonitor monitor, @NotNull DBSEntity table, Map<String, Object> options, boolean addComments) throws DBException {
         if (table instanceof DBPScriptObject) {
+            if (options == null) {
+                options = new HashMap<>();
+            }
+            options.put("needRefresh", true);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java]: Why can't you use `DBPScriptObject#OPTION_REFRESH` for that?

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreViewBase.java [hunk 2]
```diff
@@ -92,17 +89,20 @@ public String getSource() {
     @Property(hidden = true, editable = true, updatable = true, order = -1)
     public String getObjectDefinitionText(DBRProgressMonitor monitor, Map<String, Object> options) throws DBException
     {
-        if (source == null) {
+
+        boolean needRefresh = false;
+        if (options != null) {
+            needRefresh = Optional.ofNullable((Boolean) options.get("needRefresh")).orElse(false);
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreViewBase.java]:
```suggestion
            needRefresh = CommonUtils.getOption(options, "needRefresh");
```

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreViewBase.java [hunk 3]
```diff
@@ -92,17 +89,20 @@ public String getSource() {
     @Property(hidden = true, editable = true, updatable = true, order = -1)
     public String getObjectDefinitionText(DBRProgressMonitor monitor, Map<String, Object> options) throws DBException
     {
-        if (source == null) {
+
+        boolean needRefresh = false;
+        if (options != null) {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreViewBase.java:97]: It can't be `null`

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java [hunk 4]
```diff
@@ -122,6 +122,10 @@ public static String generateObjectDDL(@NotNull DBRProgressMonitor monitor, @Not
 
     public static String getTableDDL(@NotNull DBRProgressMonitor monitor, @NotNull DBSEntity table, Map<String, Object> options, boolean addComments) throws DBException {
         if (table instanceof DBPScriptObject) {
+            if (options == null) {
+                options = new HashMap<>();
+            }
+            options.put("needRefreshView", true);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/struct/DBStructUtils.java]: No need to add new constant. 
Use DBPScriptObject.OPTION_REFRESH

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreViewBase.java [hunk 5]
```diff
@@ -92,17 +94,19 @@ public String getSource() {
     @Property(hidden = true, editable = true, updatable = true, order = -1)
     public String getObjectDefinitionText(DBRProgressMonitor monitor, Map<String, Object> options) throws DBException
     {
-        if (source == null) {
+        boolean needRefresh = false;
+        if (options != null) {
+            needRefresh = Optional.ofNullable((Boolean) options.get("needRefreshView")).orElse(false);
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/PostgreViewBase.java]: Use CommonUtils.toBoolean instead

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by HocKu7** [comment]
For those who gonna test it - need to test that case as well https://github.com/dbeaver/dbeaver/issues/34267

**Review by ShadelessFox — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/PostgreUtils.java [hunk 6]
```diff
@@ -25,18 +25,38 @@
...
 import org.jkiss.dbeaver.model.DBUtils;
 import org.jkiss.dbeaver.model.connection.DBPDriver;
 import org.jkiss.dbeaver.model.edit.DBEPersistAction;
 import org.jkiss.dbeaver.model.edit.DBERegistry;
-import org.jkiss.dbeaver.model.exec.*;
+import org.jkiss.dbeaver.model.exec.DBCAttributeMetaData;
+import org.jkiss.dbeaver.model.exec.DBCEntityMetaData;
+import org.jkiss.dbeaver.model.exec.DBCException;
+import org.jkiss.dbeaver.model.exec.DBCExecutionContext;
+import org.jkiss.dbeaver.model.exec.DBCSession;
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/PostgreUtils.java]: Eh? We use `*` import after 5 imports from the same package. Please update your formatter settings.

**Comment by HocKu7** [comment]
For test team - need to test "Show permission" checkbox on objects in Postgres as well

**Comment by DariaMarkaryan** [comment]
Verified

**Pull request merged**